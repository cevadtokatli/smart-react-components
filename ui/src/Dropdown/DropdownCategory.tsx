import Div from '@smart-react-components/core/Element/Div'
import { ContentElement } from '@smart-react-components/core/types'
import React from 'react'
import { PrivateProps } from './DropdownItem'

export interface Props {
  children: JSX.Element | JSX.Element[]
  label: ContentElement
}

const DropdownCategory: React.FC<Props> = ({ children, hasHover, hasWaveEffect, isOutline, isSoft, label, palette, setStatus, waveEffectPalette }: Props & PrivateProps) => (
  <>
    <Div fontWeight="$fontWeight.dropdownCategory">{label}</Div>
    { (Array.isArray(children) ? children : [children]).map((item, idx) => React.cloneElement(item, {
      key: item.key ?? idx,
      hasHover,
      hasWaveEffect,
      isOutline,
      isSoft,
      palette,
      setStatus,
      waveEffectPalette,
    })) }
  </>
)

export default DropdownCategory
