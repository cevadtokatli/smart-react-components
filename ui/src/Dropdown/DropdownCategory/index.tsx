import Div from '@smart-react-components/core/Element/Div'
import { ContentElement } from '@smart-react-components/core/types'
import React from 'react'

export interface Props {
  children: ContentElement
}

const DropdownCategory: React.FC<Props> = ({ children }) => (
  <Div fontWeight="$fontWeight.dropdownCategory">
    {children}
  </Div>
)

export default DropdownCategory
