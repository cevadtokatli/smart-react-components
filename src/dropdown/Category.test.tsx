import React from "react"
import renderTheme from "../test/renderTheme"
import DropdownCategory from "./Category"

describe("<Dropdown />", () => {
    it("should match snapshot", () => {
        const wrapper = renderTheme(<DropdownCategory>Category</DropdownCategory>)
        expect(wrapper).toMatchSnapshot()
    })
})