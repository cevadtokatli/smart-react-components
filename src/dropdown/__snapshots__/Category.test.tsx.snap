// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> should match snapshot 1`] = `
<div
  class="sc-AxjAm sc-AxirZ dpeJfe"
  shape="default"
>
  Category
</div>
`;
